src/rubrix/client/models.py

```diff
@@ -27,6 +27,74 @@
 from rubrix.server.commons.helpers import limit_value_length
 
 
+class BaseRecord(BaseModel):
+    """Base class for our record models
+
+    Args:
+        prediction:
+            The predictions for your record.
+        annotation:
+            The annotations for your record.
+        prediction_agent:
+            Name of the prediction agent. By default, this is set to the hostname of your machine.
+        annotation_agent:
+            Name of the prediction agent. By default, this is set to the hostname of your machine.
+        id:
+            The id of the record. By default (`None`), we will generate a unique ID for you.
+        metadata:
+            Meta data for the record. Defaults to `{}`.
+        status:
+            The status of the record. Options: 'Default', 'Edited', 'Discarded', 'Validated'.
+            If an annotation is provided, this defaults to 'Validated', otherwise 'Default'.
+        event_timestamp:
+            The timestamp of the record.
+        metrics:
+            READ ONLY! Metrics at record level provided by the server when using `rb.load`.
+            This attribute will be ignored when using `rb.log`.
+    """
+
+    prediction: Optional[Any] = None
+    annotation: Optional[Any] = None
+    prediction_agent: Optional[str] = None
+    annotation_agent: Optional[str] = None
+    id: Optional[Union[int, str]] = None
+    metadata: Dict[str, Any] = Field(default_factory=dict)
+    status: Optional[str] = None
+    event_timestamp: Optional[datetime.datetime] = None
+    metrics: Optional[Dict[str, Any]] = None
+
+    @validator("metadata", pre=True)
+    def check_value_length(cls, metadata):
+        return _limit_metadata_values(metadata)
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Could we use a pydantic `@rootValidator` instead of override the `init` method? See https://pydantic-docs.helpmanual.io/usage/validators/#root-validators

Reply:
Yes, much better with `root_validators`, thanks for the tip!